samples/nrf9160/http_update/modem_full_update/src/main.c

```diff
@@ -0,0 +1,575 @@
+
+	if (valid_init) {
+		err = nrf_modem_lib_shutdown();
+		if (err != 0) {
+			printk("nrf_modem_lib_shutdown() failed: %d\n", err);
+			return err;
+		}
+	}
+
+	err = nrf_modem_lib_bootloader_init();
+	if (err != 0) {
+		printk("nrf_modem_lib_bootloader_init() failed: %d\n", err);
+		return err;
+	}
+
+	err = fmfu_fdev_load(fmfu_buf, sizeof(fmfu_buf), flash_dev, 0);
+	if (err != 0) {
+		printk("fmfu_fdev_load failed: %d\n", err);
+		return err;
+	}
+
+	err = nrf_modem_lib_shutdown();
+	if (err != 0) {
+		printk("nrf_modem_lib_shutdown() failed: %d\n", err);
+		return err;
+	}
+
+	err = nrf_modem_lib_init();
+	if (err != 0) {
+		printk("nrf_modem_lib_init() failed: %d\n", err);
+		return err;
+	}
+
+	printk("Modem firmware update completed.\n");
+
+	current_version_display();
+
+	return 0;
+}
+
+#if defined(CONFIG_USE_HTTPS)
+static int cert_provision(void)
+{
+	static const char cert[] = {
+		#include "../cert/AmazonRootCA1"
+	};
+	BUILD_ASSERT(sizeof(cert) < KB(4), "Certificate too large");
+
+	int err;
+	bool exists;
+
+	err = modem_key_mgmt_exists(TLS_SEC_TAG,
+				    MODEM_KEY_MGMT_CRED_TYPE_CA_CHAIN, &exists);
+	if (err) {
+		printk("Failed to check for certificates err %d\n", err);
+		return err;
+	}
+
+	if (exists) {
+		/* For the sake of simplicity we delete what is provisioned
+		 * with our security tag and reprovision our certificate.
+		 */
+		err = modem_key_mgmt_delete(TLS_SEC_TAG,
+					    MODEM_KEY_MGMT_CRED_TYPE_CA_CHAIN);
+		if (err) {
+			printk("Failed to delete existing certificate, err %d\n",
+			       err);
+		}
+	}
+
+	printk("Provisioning certificate\n");
+
+	/*  Provision certificate to the modem */
+	err = modem_key_mgmt_write(TLS_SEC_TAG,
+				   MODEM_KEY_MGMT_CRED_TYPE_CA_CHAIN, cert,
+				   sizeof(cert) - 1);
+	if (err) {
+		printk("Failed to provision certificate, err %d\n", err);
+		return err;
+	}
+
+	return 0;
+}
+#endif /* defined(CONFIG_USE_HTTPS) */
+
+/**
+ * @brief Configures modem to provide LTE link.
+ *
+ * Blocks until link is successfully established.
+ */
+static int modem_configure_and_connect(void)
+{
+	BUILD_ASSERT(!IS_ENABLED(CONFIG_LTE_AUTO_INIT_AND_CONNECT),
+			"This sample does not support auto init and connect");
+	int err;
+
+#if defined(CONFIG_USE_HTTPS)
+	err = cert_provision();
+	if (err) {
+		printk("Could not provision root CA to %d", TLS_SEC_TAG);
+		return err;
+	}
+
+#endif /* CONFIG_USE_HTTPS */
+
+	printk("LTE Link Connecting ...\n");
+	err = lte_lc_init_and_connect_async(lte_lc_handler);
+	if (err) {
+		printk("LTE link could not be established.");
+		return err;
+	}
+	printk("LTE Link Connected!\n");
+}
+
+void fota_dl_handler(const struct fota_download_evt *evt)
+{
+	switch (evt->id) {
+	case FOTA_DOWNLOAD_EVT_ERROR:
+		printk("Received error from fota_download\n");
+		/* Fallthrough */
+	case FOTA_DOWNLOAD_EVT_FINISHED:
+		apply_state(DFU_APPLY);
+		k_work_submit(&fota_work);
+		break;
+
+	default:
+		break;
+	}
+}
+
+static int fmfu_update_download(void)
```

Comment:
This function is called `fmfu_update_download` here, `app_update_download` and `dfu_update_start` in the other two samples.
Would it make sense to use the same name across these samples? To me it would make it easier to understand that they all have the same purpose in these samples and easier to see what is in common and what isn't.

Reply:
I can align that. I'll also align the `state enum`.